### ISSUE TITLE
Small window displacement when toggling full screen

### ISSUE THREAD

**Issue by waj** [koekeishiya/yabai#1680]
I notice small window displacements when a window full screen is toggled.

https://user-images.githubusercontent.com/22697/227733541-cd9d6959-6c85-4fd1-9246-7ba4123c449c.mov

What could be causing that displacement? Might be related but if I disable gaps between windows, sometimes adjacent windows in the layout will overlap slightly.

**yabai version**: `v5.0.2`
**macOS version**: `13.2.1 (22D68)`

Here's my config:
```
# global settings
yabai -m config mouse_follows_focus          off
yabai -m config window_shadow                on
yabai -m config window_opacity               off
yabai -m config active_window_opacity        1.0
yabai -m config normal_window_opacity        0.90
yabai -m config window_border                off
yabai -m config window_border_width          6
yabai -m config active_window_border_color   0xff775759
yabai -m config normal_window_border_color   0xff555555
yabai -m config insert_feedback_color        0xffd75f5f
yabai -m config split_ratio                  0.50
yabai -m config auto_balance                 off
yabai -m config mouse_modifier               fn
yabai -m config mouse_action1                move
yabai -m config mouse_action2                resize
yabai -m config mouse_drop_action            swap

# general space settings
yabai -m config layout                       bsp
yabai -m config top_padding                  6
yabai -m config bottom_padding               6
yabai -m config left_padding                 6
yabai -m config right_padding                6
yabai -m config window_gap                   6
```

**Comment by waj** [comment]
I did a bisect and I found it started happening after this commit: https://github.com/koekeishiya/yabai/commit/08b5422effe8bd04cf67aebc2b418b8786c8ff13

Indeed, I commented out calls to `area_ax_truncate` in master and I don't see the displacement anymore although I now see some small overlap between windows.

Something I notice is that it only happens in my second monitor. And it depends on the arrangement of the displays. So for example, it happens both in X and Y axes when the displays are arranged like this:

<img width="566" alt="image" src="https://user-images.githubusercontent.com/22697/228373386-711a8b1b-de14-4174-b42b-4c91fa312354.png">

But only happens in Y axis when the arrangement is like this:
<img width="567" alt="image" src="https://user-images.githubusercontent.com/22697/228373740-c8032f57-90d1-40e7-920a-96956b29d30b.png">
